src/common/k8s-api/kube-object.store.ts

```diff
@@ -248,29 +256,31 @@ export abstract class KubeObjectStore<T extends KubeObject> extends ItemStore<T>
     } finally {
       this.isLoading = false;
     }
+
+    return undefined;
   }
 
   @action
-  async reloadAll(opts: { force?: boolean; namespaces?: string[]; merge?: boolean } = {}) {
+  async reloadAll(opts: { force?: boolean; namespaces?: string[]; merge?: boolean } = {}): Promise<undefined | T[]> {
     const { force = false, ...loadingOptions } = opts;
 
     if (this.isLoading || (this.isLoaded && !force)) {
-      return;
+      return undefined;
     }
 
     return this.loadAll(loadingOptions);
   }
 
   @action
-  protected mergeItems(partialItems: T[], { merge = true, updateStore = true, sort = true, filter = true } = {}): T[] {
+  protected mergeItems(partialItems: T[], { merge = true, updateStore = true, sort = true, filter = true, namespaces }: MergeItemsOptions): T[] {
     let items = partialItems;
 
     // update existing items
-    if (merge) {
-      const namespaces = partialItems.map(item => item.getNs());
+    if (merge && this.api.isNamespaced) {
```

Comment:
This check is here because we shouldn't even attempt to merge cluster scoped items.